x/ibc/light-clients/07-tendermint/types/misbehaviour_handle.go

```diff
@@ -69,25 +69,14 @@ func (cs ClientState) CheckMisbehaviourAndUpdateState(
 		ageBlocks = 0
 	}
 
-	// TODO: Retrieve consensusparams from client state and not context
-	// Issue #6516: https://github.com/cosmos/cosmos-sdk/issues/6516
-	consensusParams := ctx.ConsensusParams()
-
 	// Reject misbehaviour if the age is too old. Misbehaviour is considered stale
 	// if the difference in time and number of blocks is greater than the allowed
 	// parameters defined.
-	//
-	// NOTE: The first condition is a safety check as the consensus params cannot
-	// be nil since the previous param values will be used in case they can't be
-	// retrieved. If they are not set during initialization, Tendermint will always
-	// use the default values.
-	if consensusParams != nil &&
-		consensusParams.Evidence != nil &&
-		(ageDuration > consensusParams.Evidence.MaxAgeDuration ||
-			ageBlocks > consensusParams.Evidence.MaxAgeNumBlocks) {
+	if ageDuration > cs.ConsensusParams.Evidence.MaxAgeDuration ||
```

Comment:
this will intentionally panic if the consensus params are nil which represents incorrectness in the code. We would want nodes to halt if that happened because otherwise misbehaviour could not be processed